### PR TITLE
Feature 13 resource availability

### PR DESCRIPTION
**Pull request by aleks247**

### CODE REVIEW

**Comment by Alekseeybg** [comment]
When I tried to open resource details it shows error:
image
image

File: src/components/Tables/ProductsTable.vue
Comment (VladoKat) [inline src/components/Tables/ProductsTable.vue:1]: Why have you made changes to the ProductsTable? If they are not related to the ticket, please, remove these changes.

File: src/components/Tables/ResourceTable.vue
Comment (VladoKat) [inline src/components/Tables/ResourceTable.vue:1]: Same here, if it's not part of the ticket, please, remove it.

File: src/views/Products.vue
Comment (VladoKat) [inline src/views/Products.vue:1]: Same with this, please, remove these changes if they are not related to the ticket

File: src/views/Resources.vue
Comment (VladoKat) [inline src/views/Resources.vue:1]: And here as well, please, remove if not related to the ticket

File: src/components/Tables/ResourceAvailabilityTable.vue [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+  <div>
+    <div v-if="resources.length === 0">
+      <p class="no-resources">No resources available.</p>
+    </div>
+    <div v-else>
+      <div class="table">
+        <table class="table-table">
+          <thead class="table-component">
+            <tr>
+              <th>Owner Name</th>
```
Comment (VladoKat) [inline src/components/Tables/ResourceAvailabilityTable.vue:11]: The owner name would be the same for every resource in this table because it's showing us the resources a certain user possess. That means we don't need to have it on the table rows. It might be a good idea to include it in the table header though.

File: src/components/Tables/ResourceAvailabilityTable.vue [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+              <th>Quantity</th>
+            </tr>
+          </thead>
+          <tbody class="table-component">
+            <tr v-for="(resource, index) in resources" :key="index">
+              <td>{{ resource.owner.name }}</td>
+              <td>{{ resource.resources[index].type }}</td>
+              <td>{{ resource.resources[index].clazz }}</td>
+              <td>{{ resource.resources[index].color }}</td>
+              <td>{{ resource.quantity }}</td>
```
Comment (VladoKat) [inline src/components/Tables/ResourceAvailabilityTable.vue:24]: I realise we haven't mentioned which fields to include in the rows but we might want to include all of them just like in the resources table. It's going to be overly verbose but we haven't thought of how to show enough information so the user knows which resource is meant to be shown and also to not show too much information so that they get lost in the data.
If possible some grouping of the columns might be a good idea. All the resource information columns to be in a group and the quantityType and quantity in another group.

**Review by VladoKat — CHANGES_REQUESTED** [review]
Please, remove the changes that are not necessary for this ticket. Take a look at the comments and let me know what you think.

File: src/components/Tables/ProductsTable.vue
Comment (aleks247) [reply src/components/Tables/ProductsTable.vue:1]: My bad, I was with a older version of the project where the updates were not applied, so I'm going to fix that

File: src/components/Tables/ResourceAvailabilityTable.vue (hunk 2 above)
Comment (aleks247) [reply src/components/Tables/ResourceAvailabilityTable.vue:24]: We are going to do that after the vuetify merge

Comment (aleks247) [reply src/components/Tables/ResourceAvailabilityTable.vue:24]: Actually, I made it

File: src/components/Tables/ResourceTable.vue
Comment (aleks247) [reply src/components/Tables/ResourceTable.vue:1]: Done

File: src/views/Resources.vue
Comment (aleks247) [reply src/views/Resources.vue:1]: Done

File: src/views/Products.vue
Comment (aleks247) [reply src/views/Products.vue:1]: Done

File: src/components/Tables/ResourceAvailabilityTable.vue (hunk 1 above)
Comment (aleks247) [reply src/components/Tables/ResourceAvailabilityTable.vue]: Done

File: src/components/Tables/ResourceAvailabilityTable.vue (hunk 2 above)
Comment (aleks247) [reply src/components/Tables/ResourceAvailabilityTable.vue]: Done
